doc/designs/quic-design/tx-packetiser.md

```diff
@@ -0,0 +1,327 @@
+TX Packetiser
+=============
+
+This module creates frames from the application data obtained from
+the application.  It also receives CRYPTO frames from the TLS Handshake
+Record Layer and ACK frames from the ACK Handling And Loss Detector
+subsystem.
+
+The packetiser also deals with the flow and congestion controllers.
+
+Creation & Destruction
+----------------------
+
+```c
+struct ssl_quic_tx_packetiser_st {
+    SSL *conn;
+};
+
+_owur typedef struct ssl_quic_tx_packetiser_st SSL_QUIC_TX_PACKETISER;
+
+SSL_QUIC_TX_PACKETISER ssl_quic_tx_packetiser_new(SSL *context);
+void ssl_quic_tx_packetiser_free(SSL_QUIC_TX_PACKETISER *tx);
+```
+
+Structures
+----------
+
+### Connection
+
+Represented by an SSL object.
+
+### Stream
+
+Represented by an SSL object.
+
+As per [RFC 9000 2.3 Stream Prioritization], streams should contain a priority
+provided by the calling application.  For MVP, this is not required to be
+implemented because only one stream is supported.  However, packets being
+retransmitted should be preferentially sent as noted in
+[RFC 9000 13.3 Retransmission of Information].
+
+```c
+__owur int SSL_get_stream_priority(SSL *ssl);
+```
+
+### Frame
+
+QUIC frames are represented by a leading type octet followed by the frame
+data.  As far as the packetiser is concerned, only the type byte is important:
+
```

Comment:
Worth noting the QUIC frame header is actually a variable-length integer. This is *currently* moot because all of the frame types in the QUIC RFCs are encoded in one byte, and peers aren't allowed to send extension frame types without first negotiating their support. You could change this to uint64_t, or you could leave it as is for now (I'm OK with either). Though in the latter case the text above should still be updated, and we will also need to make sure we handle large frame types when deserializing correctly.

Reply:
Yeah, I know :cry: I'm not worrying about it just yet.

Using a uint64_t leads to endian problems so I'll stick with the byte and clarify the text.